src/Codec/Xlsx/Types/DataValidation.hs

```diff
@@ -0,0 +1,200 @@
+module Codec.Xlsx.Types.DataValidation where
+
+import           Control.Applicative        ((<$>))
+import           Control.Lens.TH            (makeLenses)
+import           Control.Monad              ((>=>))
+import           Data.Char                  (isSpace)
+import           Data.Default
+import qualified Data.Map                   as M
+import           Data.Maybe                 (catMaybes)
+import           Data.Monoid                ((<>))
+import           Data.Text                  (Text)
+import qualified Data.Text                  as T
+import           Text.XML                   (Node (..), Element (..))
+import           Text.XML.Cursor            (Cursor, ($/), element)
+
+import           Codec.Xlsx.Parser.Internal
+import           Codec.Xlsx.Types.Common
+import           Codec.Xlsx.Types.ConditionalFormatting
+import           Codec.Xlsx.Writer.Internal
+
+-- See 18.18.21 "ST_DataValidationType (Data Validation Type)" (p. 2440/2450)
+data ValidationType
+    = ValidationTypeNone
+    | ValidationTypeCustom     Formula
+    | ValidationTypeDate       OperatorExpression
+    | ValidationTypeDecimal    OperatorExpression
+    | ValidationTypeList       [Text]
+    | ValidationTypeTextLength OperatorExpression
+    | ValidationTypeTime       OperatorExpression
+    | ValidationTypeWhole      OperatorExpression
+    deriving (Eq, Show)
+
+-- See 18.18.18 "ST_DataValidationErrorStyle (Data Validation Error Styles)" (p. 2438/2448)
+data ErrorStyle
+    = ErrorStyleInformation
+    | ErrorStyleStop
+    | ErrorStyleWarning
+    deriving (Eq, Show)
+
+-- See 18.3.1.32 "dataValidation (Data Validation)" (p. 1614/1624)
+data DataValidation = DataValidation
+    { _dvAllowBlank       :: Maybe Bool
+    , _dvError            :: Maybe Text
+    , _dvErrorStyle       :: Maybe ErrorStyle
+    , _dvErrorTitle       :: Maybe Text
+    , _dvPrompt           :: Maybe Text
+    , _dvPromptTitle      :: Maybe Text
+    , _dvShowDropDown     :: Maybe Bool
+    , _dvShowErrorMessage :: Maybe Bool
+    , _dvShowInputMessage :: Maybe Bool
+    , _dvSqref            :: SqRef
+    , _dvValidationType   :: Maybe ValidationType
+    } deriving (Eq, Show)
+
+makeLenses ''DataValidation
+
+instance Default DataValidation where
+    def = DataValidation
+      Nothing Nothing Nothing Nothing Nothing Nothing Nothing Nothing Nothing
+      (SqRef []) Nothing
+
+{-------------------------------------------------------------------------------
+  Parsing
+-------------------------------------------------------------------------------}
+
+instance FromAttrVal ErrorStyle where
+    fromAttrVal "information" = readSuccess ErrorStyleInformation
+    fromAttrVal "stop"        = readSuccess ErrorStyleStop
+    fromAttrVal "warning"     = readSuccess ErrorStyleWarning
+    fromAttrVal t             = invalidText "ErrorStyle" t
+
+instance FromCursor DataValidation where
+    fromCursor cur = cur $/ element (n_ "dataValidation") >=> fromCur
```

Comment:
that doesn't look correct - `fromCursor` is supposed to be invoked with cursor pointing to the node which should give you results, i.e. it should already point to `<dataValidation>`, i.e. you miss that level in `Parser`